tests/uc_oss/test_uc_oss_integration.py

```diff
@@ -0,0 +1,152 @@
+        subprocess.Popen(
+            [sys.executable, "-m", "mlflow", "server", "--port", str(port)]
+        ) as mlflow_proc,
+    ):
+        try:
+            _await_server_up_or_die(int(port))
+            _await_server_up_or_die(8080)
+
+            mlflow_tracking_url = f"http://127.0.0.1:{port}"
+            uc_oss_url = "uc:http://127.0.0.1:8080"
+
+            mlflow.set_tracking_uri(mlflow_tracking_url)
+            mlflow.set_registry_uri(uc_oss_url)
+
+            yield mlflow_tracking_url
+        finally:
+            mlflow_proc.terminate()
+            uc_proc.terminate()
+
+
+def test_integration(setup_servers):
+    catalog = "unity"
+    schema = "default"
+    registered_model_name = "iris"
+    model_name = f"{catalog}.{schema}.{registered_model_name}"
+    mlflow.set_experiment("iris-uc-oss")
+    client = mlflow.MlflowClient()
+    with pytest.raises(Exception, match="NOT_FOUND"):
+        client.get_registered_model(model_name)
+
+    X, y = datasets.load_iris(return_X_y=True, as_frame=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+
+    def build_model():
+        with mlflow.start_run():
+            # Train a sklearn model on the iris dataset
+            clf = RandomForestClassifier(max_depth=7)
+            clf.fit(X_train, y_train)
+            # Take the first row of the training dataset as the model input example.
+            input_example = X.iloc[[0]]
+            # Log the model and register it as a new version in UC.
+            mlflow.sklearn.log_model(
+                clf,
+                "model",
+                # The signature is automatically inferred from the input example and
+                # its predicted output.
+                input_example=input_example,
+                registered_model_name=model_name,
+            )
+
+    build_model()
+    model_version = 1
+    model_uri = f"models:/{model_name}/{model_version}"
+    rm_desc = "UC-OSS/MLflow Iris model"
+    mv_desc = "Version 1 of the UC-OSS/MLflow Iris model"
+
+    # Load the model and do some batch inference.
+    # By specifying the UC OSS model uri, mlflow will make UC OSS
+    # REST API calls to retrieve the model
+    loaded_model = mlflow.pyfunc.load_model(model_uri)
+    predictions = loaded_model.predict(X_test)
+    iris_feature_names = datasets.load_iris().feature_names
+    result = pd.DataFrame(X_test, columns=iris_feature_names)
+    result["actual_class"] = y_test
+    result["predicted_class"] = predictions
+    assert result[:4] is not None
+
+    # list_artifacts will use the UC OSS model URI and make REST API calls to
+    # UC OSS to:
+    #   1) retrieve credentials (none for file based UC OSS)
+    #   2) use the storage location returned from UC OSS for the model version
+    #      list the artifacts stored in the location
+    mlflow.artifacts.list_artifacts(model_uri)
+
+    path = os.path.join("/tmp", "models", model_name, str(model_version))
```

Comment:
Let's use the tmp_path fixture.

Reply:
/ai give us a minimum example demonstrating how to use the pytest's tmp_path fixture.

Reply:
Thanks, @kriscon-db We should use `tmp_path` in tests to avoid leaving unnecessary files/directories.